### PR TITLE
Check osm-controller proxy metrics

### PR DESCRIPTION
**Pull request by johnsonshi**
Check proxy metrics returned from
osm-controller's http server.

### CODE REVIEW

**Comment by johnsonshi** [comment]
Sample failing check for osm_proxy_connect_count metric:

image

Sample successful check for osm_proxy_connect_count metric:

image

**Comment by johnsonshi** [comment]
Had to move files to separate packages to resolve circular imports.

File: pkg/osm/controller/health.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	return nil
+}
+
+func checkControllerHealthLiveness(controllerHTTPServerURL string) error {
+	url := fmt.Sprintf("%s%s", controllerHTTPServerURL, httpserverconstants.HealthLivenessPath)
+	if err := utils.CheckHTTPResponseCodeEquals(url, http.StatusOK); err != nil {
+		return errors.Errorf("osm-controller health liveness check failed: %s", err)
+	}
+	return nil
+}
```
Comment (johnsonshi) [inline pkg/osm/controller/health.go:27]: No changes - just migrated the code to another file (health.go).

File: pkg/osm/controller/health_test.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+			}))
+			defer ts.Close()
+
+			err := checkControllerHealthReadiness(ts.URL)
+			assert.Equal(t, test.shouldError, err != nil)
+			err = checkControllerHealthLiveness(ts.URL)
+			assert.Equal(t, test.shouldError, err != nil)
+		})
+	}
+}
```
Comment (johnsonshi) [inline pkg/osm/controller/health_test.go:42]: No changes - just migrated the code.

File: pkg/osm/controller/httpserver.go [hunk 3]
```diff
@@ -101,19 +103,3 @@ func (check HTTPServerEndpointsCheck) Suggestion() string {
...
-	return nil
-}
-
-func checkControllerHealthLiveness(controllerHTTPServerURL string) error {
-	url := fmt.Sprintf("%s%s", controllerHTTPServerURL, httpserverconstants.HealthLivenessPath)
-	if err := utils.CheckHTTPResponseCodeEquals(url, http.StatusOK); err != nil {
-		return errors.Errorf("osm-controller health liveness check failed: %s", err)
-	}
-	return nil
-}
```
Comment (johnsonshi) [inline pkg/osm/controller/httpserver.go:119]: No changes - just migrated the code.

File: pkg/connectivity/pod-to-pod.go [hunk 4]
```diff
@@ -11,7 +11,7 @@ import (
 	"github.com/openservicemesh/osm-health/pkg/kubernetes/namespace"
 	"github.com/openservicemesh/osm-health/pkg/kubernetes/pod"
 	"github.com/openservicemesh/osm-health/pkg/kubernetes/podhelper"
-	"github.com/openservicemesh/osm-health/pkg/osm"
+	"github.com/openservicemesh/osm-health/pkg/osm/utils"
```
Comment (draychev) [inline pkg/connectivity/pod-to-pod.go:14]: I'm typically not a huge fan of using packages like `utils` - it becomes the `misc` package.
Not a big deal, but there's got to be a better place for this. Seems like `osm` was an ok fit too.

File: pkg/osm/version/annotations.go [hunk 5]
```diff
@@ -1,4 +1,4 @@
-package osm
+package version
```
Comment (draychev) [inline pkg/osm/version/annotations.go:1]: What's the reason behind this rename?
It seems that these are `osm` CRDs, no?

File: pkg/osm/utils/utils.go [hunk 6]
```diff
@@ -70,3 +73,68 @@ func FormatReleaseVersion(version string) (string, error) {
...
+func GetMonitoredNamespaces(client kubernetes.Interface, osmControlPlaneNamespace common.MeshNamespace) (*corev1.NamespaceList, error) {
+	meshInfo, err := GetMeshInfo(client, osmControlPlaneNamespace)
+	if err != nil {
+		return &corev1.NamespaceList{}, err
+	}
+	meshName := meshInfo.Name
+
+	// Criteria that is used to determine if a namespace is monitored by osm (from osm's mutating webhook):
+	// 		kubectl get MutatingWebhookConfiguration osm-webhook-osm -o json | jq '.webhooks[0].namespaceSelector'
+	/*
```
Comment (draychev) [inline pkg/osm/utils/utils.go:87]: Thanks for documenting this!

**Review by draychev — APPROVED** [review]

File: pkg/osm/version/annotations.go (hunk 5 above)
Comment (johnsonshi) [reply pkg/osm/version/annotations.go:1]: @draychev - I had to move some stuff from `pkg/osm` to `pkg/osm/utils` and `pkg/osm/version` as that was the only way to resolve circular dependencies.

Comment (johnsonshi) [reply pkg/osm/version/annotations.go:1]: @draychev - Let me know if these package renames are good to go or if there's another better way to resolve circular imports.

**Pull request merged**